### ISSUE TITLE
[Bug]: [Nightly] Search failed showing "fail to search on all shard leaders, err=fail to Search, QueryNode ID=3, reason=Search 3 failed, reason cannot find sealed segment 434204011045847042 in QueryNode"

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#17850]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 797218a
- Deployment mode(standalone or cluster): standalone
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed showing "fail to search on all shard leaders, err=fail to Search, QueryNode ID=3, reason=Search 3 failed, reason cannot find sealed segment 434204011045847042 in QueryNode"

### Expected Behavior

NO error

### Steps To Reproduce

Running nightly:
https://ci.milvus.io:18080/jenkins/blue/organizations/jenkins/milvus-nightly-ci/detail/master/620/pipeline

```
[2022-06-27T19:39:34.689Z] utils/wrapper.py:33: in inner_wrapper

[2022-06-27T19:39:34.689Z]     result = func(*args, **kwargs)

[2022-06-27T19:39:34.689Z] base/collection_wrapper.py:153: in search

[2022-06-27T19:39:34.689Z]     timeout=timeout, **kwargs).run()

[2022-06-27T19:39:34.689Z] check/func_check.py:56: in run

[2022-06-27T19:39:34.689Z]     result = self.check_search_results(self.response, self.func_name, self.check_items)

[2022-06-27T19:39:34.689Z] check/func_check.py:205: in check_search_results

[2022-06-27T19:39:34.689Z]     search_res = search_res.result()

[2022-06-27T19:39:34.689Z] /usr/local/lib/python3.6/site-packages/pymilvus/orm/future.py:30: in result

[2022-06-27T19:39:34.689Z]     return self.on_response(self._f.result())

[2022-06-27T19:39:34.690Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:185: in result

[2022-06-27T19:39:34.690Z]     self.exception()

[2022-06-27T19:39:34.690Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:242: in exception

[2022-06-27T19:39:34.690Z]     raise self._exception

[2022-06-27T19:39:34.690Z] /usr/local/lib/python3.6/site-packages/pymilvus/client/asynch.py:232: in done

[2022-06-27T19:39:34.690Z]     self._results = self.on_response(self._response)

[2022-06-27T19:39:34.690Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

[2022-06-27T19:39:34.690Z] 

[2022-06-27T19:39:34.690Z] self = <pymilvus.client.asynch.ChunkedSearchFuture object at 0x7f4c08d832e8>

[2022-06-27T19:39:34.690Z] response = [status {

[2022-06-27T19:39:34.690Z]   error_code: UnexpectedError

[2022-06-27T19:39:34.690Z]   reason: "fail to search on all shard leaders, err=fail to Search, QueryNode ID=3, reason=Search 3 failed, reason cannot find sealed segment 434204011045847042 in QueryNode err %!w(<nil>)"

[2022-06-27T19:39:34.690Z] }

[2022-06-27T19:39:34.690Z] ]

[2022-06-27T19:39:34.690Z] 

[2022-06-27T19:39:34.690Z]     def on_response(self, response):

[2022-06-27T19:39:34.690Z]         for raw in response:

[2022-06-27T19:39:34.690Z]             if raw.status.error_code != 0:

[2022-06-27T19:39:34.690Z] >               raise MilvusException(raw.status.error_code, raw.status.reason)

[2022-06-27T19:39:34.690Z] E               pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=3, reason=Search 3 failed, reason cannot find sealed segment 434204011045847042 in QueryNode err %!w(<nil>))>

[2022-06-27T19:39:34.690Z] 
```

### Milvus Log

1. milvus logs:
[artifacts-milvus-standalone-master-620-pymilvus-e2e-logs.tar.gz](https://ci.milvus.io:18080/jenkins/job/milvus-nightly-ci/job/master/620/artifact/artifacts-milvus-standalone-master-620-pymilvus-e2e-logs.tar.gz)
2. collection name: search_collection_h6RVOskj
3. Case failed time:
[2022-06-27T19:07:38.703Z] [gw1] [ 54%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_collection_after_release_load[3000-500-8-True-True]

**Comment by binbinlv** [comment]
And I found there is a panic from log, do not confirm whether there is any connections with this issue, paste here for some info.

ms-620-n-milvus-standalone-557d7d6d57-s9jpv_milvus-ci_standalone-b044b606b32b5e56538c15589218b29e053f3bb9a5632effef934555027c3236.log:

```

[2022/06/27 19:39:46.196 +00:00] [ERROR] [shard_node_detector.go:154] ["failed to handle watch node error"] [error="rpc error: code = Canceled desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp: lookup ms-620-n-etcd on 10.201.0.10:53: server misbehaving\""] [stack="github.com/milvus-io/milvus/internal/querynode.(*etcdShardNodeDetector).watch\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_node_detector.go:154"]

[2022/06/27 19:39:46.196 +00:00] [ERROR] [shard_segment_detector.go:132] ["failed to handle watch segment error, panic"] [error="rpc error: code = Canceled desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp: lookup ms-620-n-etcd on 10.201.0.10:53: server misbehaving\""] [stack="github.com/milvus-io/milvus/internal/querynode.(*etcdShardSegmentDetector).watch\n\t/go/src/github.com/milvus-io/milvus/internal/querynode/shard_segment_detector.go:132"]

panic: rpc error: code = Canceled desc = latest balancer error: last connection error: connection error: desc = "transport: Error while dialing dial tcp: lookup ms-620-n-etcd on 10.201.0.10:53: server misbehaving"



goroutine 1944231 [running]:

github.com/milvus-io/milvus/internal/querynode.(*etcdShardNodeDetector).watch(0xc004acaaf0, 0xc014fe8480, 0x6069a753dcc0001, 0x6069a19341025d6)

        /go/src/github.com/milvus-io/milvus/internal/querynode/shard_node_detector.go:155 +0x8c5

created by github.com/milvus-io/milvus/internal/querynode.(*etcdShardNodeDetector).watchNodes

        /go/src/github.com/milvus-io/milvus/internal/querynode/shard_node_detector.go:123 +0x10d1

panic: rpc error: code = Canceled desc = latest balancer error: last connection error: connection error: desc = "transport: Error while dialing dial tcp: lookup ms-620-n-etcd on 10.201.0.10:53: server misbehaving"



goroutine 493295 [running]:

github.com/milvus-io/milvus/internal/querynode.(*etcdShardSegmentDetector).watch(0xc00699ef50, 0xc00127d8c0, 0x6069a3421dc0001, 0x6069a1934100837, 0xc009f83020, 0x2d)

        /go/src/github.com/milvus-io/milvus/internal/querynode/shard_segment_detector.go:133 +0x8aa

created by github.com/milvus-io/milvus/internal/querynode.(*etcdShardSegmentDetector).watchSegments

        /go/src/github.com/milvus-io/milvus/internal/querynode/shard_segment_detector.go:108 +0xb71
```

**Comment by binbinlv** [comment]
I found there is an issue reporting the similar error, I am wondering whether the root cause is the same or not?

https://github.com/milvus-io/milvus/issues/17347

**Comment by binbinlv** [comment]
@xiaofan-luan Could you help to assign anyone to check this issue? Because this is the latest search issue, so set it as urgent first.

**Comment by yah01** [comment]
/assign @binbinlv 
fixed with #17950 #18020

**Comment by binbinlv** [comment]
It has not been reproduced these days and fixed, so close it.
